x/concentrated-liquidity/incentives_test.go

```diff
@@ -3509,3 +3511,241 @@ func (s *KeeperTestSuite) TestPrepareBalancerPoolAsFullRange() {
+			}
+
+			// Link the balancer and CL pools
+			s.App.GAMMKeeper.SetMigrationInfo(s.Ctx,
+				gammtypes.MigrationRecords{
+					BalancerToConcentratedPoolLinks: []gammtypes.BalancerToConcentratedPoolLink{
+						{BalancerPoolId: balancerPoolId, ClPoolId: clPoolId},
+					},
+				})
+
+			// Add balancer shares to CL accumulatores
+			addedLiquidity := sdk.ZeroDec()
+			if !tc.balSharesNotAddedToAccums {
+				addedBalPool, qualifiedShares, err := s.App.ConcentratedLiquidityKeeper.PrepareBalancerPoolAsFullRange(s.Ctx, clPool.GetId())
+				addedLiquidity = addedLiquidity.Add(qualifiedShares)
+
+				// If a valid link exists, ensure no error and sanity check the output pool ID
+				if !tc.concentratedPoolDoesNotExist && !tc.balancerPoolDoesNotExist {
+					s.Require().NoError(err)
+					s.Require().Equal(addedBalPool, balancerPoolId)
+				}
+			}
+
+			// Emit incentives to the uptime accumulators
+			for _, growth := range tc.uptimeGrowth {
+				decEmissions := growth.MulDec(initialLiquidity.Add(addedLiquidity))
+				normalizedEmissions := sdk.NormalizeCoins(decEmissions)
+				s.FundAcc(clPool.GetIncentivesAddress(), normalizedEmissions)
+			}
+			addToUptimeAccums(s.Ctx, clPool.GetId(), s.App.ConcentratedLiquidityKeeper, tc.uptimeGrowth)
+
+			// --- System under test ---
+
+			amountClaimed, err := s.App.ConcentratedLiquidityKeeper.ClaimAndResetFullRangeBalancerPool(s.Ctx, clPoolId, balancerPoolId)
+
+			// --- Assertions ---
+
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
+				s.Require().Equal(sdk.Coins{}, amountClaimed)
+
+				clPoolUptimeAccumulators, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, clPool.GetId())
+				s.Require().NoError(err)
+
+				for _, uptimeAccum := range clPoolUptimeAccumulators {
+					currAccumShares, err := uptimeAccum.GetTotalShares()
+					s.Require().NoError(err)
+
+					// Ensure accum shares remain unchanged after error
+					s.Require().Equal(initialLiquidity.Add(addedLiquidity), currAccumShares)
+				}
+
+				// If gauge exists, ensure it remains empty after error
+				if !tc.balancerPoolDoesNotExist {
+					gaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, balancerPoolId, longestLockableDuration)
+					s.Require().NoError(err)
+
+					gauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, gaugeId)
+					s.Require().NoError(err)
+
+					s.Require().Equal(sdk.Coins(nil), gauge.Coins)
+				}
+
+				// Pool liquidity should remain unchanged
+				updatedClPool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, clPool.GetId())
+				s.Require().NoError(err)
+				s.Require().Equal(initialLiquidity, updatedClPool.GetLiquidity())
+
+				return
+			}
+
+			s.Require().NoError(err)
+
+			clPoolUptimeAccumulators, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, clPool.GetId())
+			s.Require().NoError(err)
+
+			for _, uptimeAccum := range clPoolUptimeAccumulators {
```

Comment:
Let's add a check that `clPoolUptimeAccumulators` are not empty so that  assertions inside the loop aren't accidentally skipped

Reply:
`clPoolUptimeAccumulators` can never be empty. Will add this check, but in general I think adding too many extraneous checks make tests much less readable/can be misleading towards the functionality of these components (e.g. someone might assume that it is possible for `clPoolUptimeAccumulators` to be empty since we check it)

Reply:
> clPoolUptimeAccumulators can never be empty. 

Thanks for adding it. I understand that this is the case today, and, currently, it does not make sense for them to be empty.

However, as the system evolves a lot of unexpected changes can happen. A few months from now,  I might not remember this guarantee. So a check like this would help me be confident that the assertions in the loop are running with no additional manual work.

I've also been noticing a lot of instances where we get false positive passes due to the lack of such checks so I definitely think it's useful.